inc/global_assets/namespace.php

```diff
@@ -0,0 +1,343 @@
+	add_action( 'init', __NAMESPACE__ . '\\maybe_create_site' );
+
+	// Redirect media site URLs.
+	add_action( 'admin_init', __NAMESPACE__ . '\\redirect_dashboard' );
+	add_action( 'template_redirect', __NAMESPACE__ . '\\redirect_frontend' );
+
+	// Handle clean up operations.
+	add_action( 'wp_uninitialize_site', __NAMESPACE__ . '\\uninitialize_media_site' );
+
+	// Handle global media library admin customisations.
+	add_action( 'admin_menu', __NAMESPACE__ . '\\admin_menu', 1000 );
+	add_action( 'admin_bar_menu', __NAMESPACE__ . '\\admin_bar_menu', 1000 );
+
+	// Handle media site URL changes.
+	add_action( 'updated_option_siteurl', __NAMESPACE__ . '\\handle_siteurl_update', 10, 2 );
+	add_action( 'updated_option_home', __NAMESPACE__ . '\\handle_siteurl_update', 10, 2 );
+	add_action( 'wp_update_site', __NAMESPACE__ . '\\handle_site_update' );
+
+	// Do not allow media site deletion.
+	add_filter( 'map_meta_cap', __NAMESPACE__ . '\\prevent_site_deletion', 10, 4 );
+
+	// Handle network admin sites list.
+	add_filter( 'manage_sites_action_links', __NAMESPACE__ . '\\sites_list_row_actions', 10, 2 );
+}
+
+/**
+ * Returns true if the current site is the global media site.
+ *
+ * @param int|null $site_id An optional site ID to check. Defaults to the current site.
+ * @return boolean
+ */
+function is_media_site( ?int $site_id = null ) : bool {
+	return ! empty( get_site_meta( $site_id ?? get_current_blog_id(), 'is_media_site' ) );
+}
+
+/**
+ * Load Asset Mnnager Framework.
+ *
+ * @return void
+ */
+function load_amf() {
+	$config = Altis\get_config()['modules']['media'];
+
+	// Load Asset Manager Framework.
+	require_once Altis\ROOT_DIR . '/vendor/humanmade/asset-manager-framework/plugin.php';
+
+	// Load WP AMF and configure global site.
+	if ( $config['global-media-library'] ) {
+		require_once Altis\ROOT_DIR . '/vendor/humanmade/amf-wordpress/plugin.php';
+	}
+}
+
+/**
+ * Create the Global Media Library site.
+ *
+ * @throws Exception If media site cannot be created.
+ */
+function maybe_create_site() {
+	// Wait until post installation.
+	if ( defined( 'WP_INITIAL_INSTALL' ) && WP_INITIAL_INSTALL ) {
+		return;
+	}
+
+	$media_site = get_site_option( 'global_media_site' );
+	if ( ! empty( $media_site ) ) {
+		define( 'AMF_WORDPRESS_URL', $media_site );
+		return;
+	}
+
+	try {
+		$media_site_error = get_site_option( 'global_media_site_error' );
+		if ( ! empty( $media_site_error ) ) {
+			throw new Exception( $media_site_error );
+		}
+
+		/**
+		 * Filters the args used to create the global media site.
+		 *
+		 * @param array $media_site_args The arguments array for creating the global media site.
+		 */
+		$media_site_args = apply_filters( 'altis.media.global_site_args', [
+			'domain' => wp_parse_url( home_url(), PHP_URL_HOST ),
+			'path' => '/media/',
+			'title' => __( 'Global Media Library', 'altis' ),
+			'user_id' => get_user_by( 'login', get_super_admins()[0] )->ID,
+			'meta' => [
+				'is_media_site' => true,
```

Comment:
How does this meta value get assigned to a site that was manually defined as the global media library? As it stands, it looks like a manually-defined global media library URL does not get the meta value, and so is never seen as `is_media_site`.

Reply:
Correct, we're going to handle that internally so if you're using a custom site URL it's out of our hands.